widget/list.go

```diff
@@ -390,15 +397,88 @@ func (l *List) createWidget() {
 			SliderOpts.Direction(DirectionHorizontal),
 			SliderOpts.MinMax(0, 1000),
 			SliderOpts.PageSizeFunc(func() int {
-				return int(math.Round(float64(l.scrollContainer.ContentRect().Dx()) / float64(content.GetWidget().Rect.Dx()) * 1000))
+				return int(math.Round(float64(l.scrollContainer.ContentRect().Dx()) / float64(l.listContent.GetWidget().Rect.Dx()) * 1000))
 			}),
 			SliderOpts.ChangedHandler(func(args *SliderChangedEventArgs) {
 				l.scrollContainer.ScrollLeft = float64(args.Slider.Current) / 1000
 			}),
 		}...)...)
 		l.container.AddChild(l.hSlider)
 	}
-	l.sliderOpts = nil
+
+}
+
+func (l *List) SetEntries(entries []interface{}) {
+	l.entries = entries
+	l.entriesButtonTable = make(map[interface{}]*Button, len(entries))
+	l.idEntryTable = make(map[string]interface{}, len(entries))
+	l.container.RemoveChildren()
+	l.createWidget()
+	l.resetFocusIndex()
+}
+
+func (l *List) RemoveEntry(entry interface{}) {
+	if l.entryIdentifier == nil {
+		log.Println("cannot add entry without identifier function")
+		return
+	}
+	l.init.Do()
+
+	identifier := (*l.entryIdentifier)(entry)
+	targetEntry := l.idEntryTable[identifier]
+	if len(l.entries) > 1 {
+		for i, e := range l.entries {
+			if e == targetEntry {
```

Comment:
Can you please explain the entryIdentifier thing? It doesnt really seem like it buys us anything since we're looping through the entries anyways? Why do we want/need this?

Reply:
Hey, basically when removing an entry it is a copy of the value so when iterating over entries the == comparison is false, since it is a copy, if we save it on a map it is the same value so comparison is true we iterate over all of the array to be able to remove the entry from the array. Open to any suggestion.